sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Rules/AssertionArgsShouldBePassedInCorrectOrderTest.cs

```diff
@@ -19,24 +19,49 @@
  */
 
 extern alias csharp;
-using Microsoft.VisualStudio.TestTools.UnitTesting;
 using csharp::SonarAnalyzer.Rules.CSharp;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
 
 namespace SonarAnalyzer.UnitTest.Rules
 {
     [TestClass]
     public class AssertionArgsShouldBePassedInCorrectOrderTest
     {
-        [TestMethod]
+        [DataTestMethod]
+        [DataRow("1.1.11")]
+        [DataRow(AssemblyReference.NuGetInfo.LatestVersion)]
+        [TestCategory("Rule")]
+        public void AssertionArgsShouldBePassedInCorrectOrder_MsTest(string testFwkVersion)
+        {
+            Verifier.VerifyAnalyzer(@"TestCases\AssertionArgsShouldBePassedInCorrectOrder.MsTest.cs",
+                new AssertionArgsShouldBePassedInCorrectOrder(),
+                null,
+                AssemblyReference.FromNuGet("Microsoft.VisualStudio.TestPlatform.TestFramework.dll", "MSTest.TestFramework", testFwkVersion));
+        }
+
+        [DataTestMethod]
+        [DataRow("2.5.7.10213")]
+        [DataRow(AssemblyReference.NuGetInfo.LatestVersion)]
+        [TestCategory("Rule")]
+        public void AssertionArgsShouldBePassedInCorrectOrder_NUnit(string testFwkVersion)
+        {
+            Verifier.VerifyAnalyzer(@"TestCases\AssertionArgsShouldBePassedInCorrectOrder.NUnit.cs",
+                new AssertionArgsShouldBePassedInCorrectOrder(),
+                null,
+                AssemblyReference.FromNuGet("nunit.framework.dll", "NUnit", testFwkVersion));
+        }
+
+        [DataTestMethod]
+        [DataRow("2.0.0")]
+        [DataRow(AssemblyReference.NuGetInfo.LatestVersion)]
         [TestCategory("Rule")]
-        public void AssertionArgsShouldBePassedInCorrectOrder()
+        public void AssertionArgsShouldBePassedInCorrectOrder_XUnit(string testFwkVersion)
         {
-            Verifier.VerifyAnalyzer(@"TestCases\AssertionArgsShouldBePassedInCorrectOrder.cs",
+            Verifier.VerifyAnalyzer(@"TestCases\AssertionArgsShouldBePassedInCorrectOrder.Xunit.cs",
                 new AssertionArgsShouldBePassedInCorrectOrder(),
                 null,
-                Verifier.MicrosoftVisualStudioTestToolsUnitTestingAssembly,
-                Verifier.NUnitFrameworkAssembly,
-                Verifier.XunitAssertAssembly);
+                AssemblyReference.FromNuGet("xunit.assert.dll", "xunit.assert", testFwkVersion),
```

Comment:
I guess we use relatively few nugets, so we could have some convenience methods:
```
NugetReference.XUnit(string version = NugetInfo.LatestVersion) { ... }
NugetReference.MsTest(string version = NugetInfo.LatestVersion) { ... }
...
FrameworkReference.Mscorlib() => AssemblyReference.FromFramework(...);
FrameworkReference.SystemComponentModelComposition() { ... }
// or properties
FrameworkReference.Mscorlib { get; } = AssemblyReference.FromFramework(...);
```

Reply:
I was thinking about having some `enum` for the framework assemblies as the list is finite. Regarding NuGet we could probably improve the usage even more, I agree.